edb/ir/staeval.py

```diff
@@ -422,10 +422,14 @@ def cast_const_to_python(ir: irast.TypeCast, schema: s_schema.Schema) -> Any:
     sval = evaluate_to_python_val(ir.expr, schema=schema)
     if sval is None:
         return None
-    elif isinstance(sval, tuple):
-        return tuple(pytype(elem) for elem in sval)
+    if pytype is bool and isinstance(sval, str):
+        conv = lambda v: v.lower() == 'true'
     else:
-        return pytype(sval)
+        conv = pytype
+    if isinstance(sval, tuple):
```

Comment:
I doubt if this `tuple` branch is still in use 🤔 so this can be:

```python
    if isinstance(sval, tuple):
        if pytype is bool:
            return tuple(
                (
                    (elem.lower() == "true")
                    if isinstance(elem, str)
                    else bool(elem)
                )
                for elem in sval
            )
        else:
            return tuple(pytype(elem) for elem in sval)
    else:
        if pytype is bool and isinstance(sval, str):
            return sval.lower() == 'true'
        else:
            return pytype(sval)
```

or with cleaner `singledispatch` 🤷‍♂️ if `tuple` is still in use.